### ISSUE TITLE
DeepLinkConfigToken did not return valid data

### ISSUE THREAD

**Issue by jetav8r** [ionic-team/ionic-app-scripts#1106]
**Note: for support questions, please use one of these channels:**

https://forum.ionicframework.com/
http://ionicworldwide.herokuapp.com/

#### Short description of the problem:
when running the production build, it exits at deeplinks... 

#### What behavior are you expecting?
To be able to build a production release

**Steps to reproduce:**
1.run ionic cordova build browser --prod --verbose
2.
3.

```
insert any relevant code between the above and below backticks
```

**Which @ionic/app-scripts version are you using?**
I've tried several versions...right now I'm using versions of devDependencies below:
```
"devDependencies": {
    "@ionic/app-scripts": "^2.0.1-201707121621",
    "@ionic/cli-plugin-cordova": "1.5.0-alpha.18371e18",
    "@ionic/cli-plugin-ionic-angular": "1.4.0-alpha.18371e18",
    "typescript": "2.3.4"
  },
```

**Other information:** (e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links, forum links, etc)

Here's the console output after running the build command:

[16:21:50]  build prod started ...
[16:21:50]  clean started ...
[16:21:50]  clean finished in less than 1 ms
[16:21:50]  copy started ...
[16:21:50]  ngc started ...
[16:21:59]  ngc finished in 9.48 s
[16:22:00]  preprocess started ...
[16:22:00]  deeplinks started ...
[WARN] Error occurred during command execution from a CLI plugin (@ionic/cli-plugin-cordova).
[DEBUG] !!! ERROR ENCOUNTERED !!!
Error: The RegExp to find the DeepLinkConfigToken did not return valid data
[DEBUG] Error: The RegExp to find the DeepLinkConfigToken did not return valid data
        at new BuildError
        (C:\Users\me\WebstormProjects\app\node_modules\@ionic\app-scripts\dist\util\errors.js:16:28)
        at C:\Users\me\WebstormProjects\app\node_modules\@ionic\app-scripts\dist\preprocess.js:18:21

**Comment by danbucholtz** [comment]
@jetav8r,

What Angular version are you running?

Thanks,
Dan

**Comment by jetav8r** [comment]
4.1.3... I was able to get it to work by downgrading from angular 4.2
Didn't realize I had to remove the ^ symbol  in the package.json file in
order for it to force the downgrade back to 4.1.3
It's working now that angular is back to 4.1.3

Thanks for the email!

Wayne

**Comment by danbucholtz** [comment]
@jetav8r,

Yep, Angular 4.2 changed something in the ng factory files. We'll need to upgrade to match them.

Thanks,
Dan

**Comment by antoineol** [comment]
I'm also having this issue, and it seems some browsers have issues with animations below Angular 4.2 (like https://github.com/angular/material2/issues/5019, https://github.com/angular/angular/issues/14007 and I'm also having a `Partial keyframes are not supported` I cannot explain on an older browser). Supporting Angular 4.2+ would be great to limit the risk of issues with things like animations.

Any update so far about this issue?

Thanks!

**Comment by dev-manager-uk** [comment]
@danbucholtz We are getting this issue too.  Any eta on a fix?  Thanks!

**Comment by cmartin81** [comment]
I had a look inside the sourcecode. And the appNgModuleFactoryFileContent has changed. It cannot find any matches on the regex /this.*?DeepLinkConfigToken.*?=([\s\S]*?);/g
This only happens when you run it with the --aot or --prod option

**Comment by soumak77** [comment]
@danbucholtz has the Ionic team run into this issue yet now that ionic-angular uses angular 4.3.6?  I'm running the latest ionic-angular build and I'm hitting this issue now that angular has been upgraded.

**Comment by soumak77** [comment]
@danbucholtz seems like this issue is resolved.  I just tested with `@ionic/app-scripts@nightly` and the production build succeeded!

**Comment by pedrohills** [comment]
npm install @ionic/app-scripts@nightly --save-dev worked for me.

**Comment by menorval** [comment]
Thanks @pedrohills, it also worked for me.
